grammarField.js

```diff
@@ -35,55 +35,81 @@ class GrammarField {
     this.featureMatrix = math.sparse(math.zeros(maxSize, 21));
 
     this.calculate(sentences, times)
+    
+    
   }
-
+  
+  
+  
+  /**
+   * singularizes all the words in the blacklist and converts them to lowercase
+   */
+  preprocessWordBlacklist(bl=[])
+  {
+    var newBL=[];
+    for(let i=0;i<bl.length;i++)
```

Comment:
Add a space after for

Add spaces between all operators (=, <, etc)

Add spaces after the ;